packages/node_modules/@ciscospark/internal-plugin-metrics/src/metrics.js

```diff
@@ -99,6 +114,107 @@ const Metrics = SparkPlugin.extend({
       eventPayload: payload
     };
     return this.callDiagnosticEventsBatcher.request(event);
+  },
+
+  storeMetric(id, metricProps, metricType) {
+    const tuple = {metric: metricProps, type: metricType};
+    return this.metricMap.set(id, tuple);
+  },
+
+  getMetrics() {
+    return this.metricMap;
+  },
+
+  getMetric(id) {
+    return this.metricMap.get(id);
+  },
+
+  deleteMetric(id) {
+    return this.metricMap.delete(id);
+  },
+
+  registerMetric(eventName, id, props, type) {
+    if (!this.getMetric(id)) {
+      // hash name and props into table/object
+      const payload = this.createPayload(eventName, props);
+      return this.storeMetric(id, payload, type);
+    }
+    return false;
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
+  */
+  clientRegisterMetric(eventName, id, props) {
+    return this.registerMetric(eventName, id, props, this.CLIENT_METRIC);
+  },
+
+  /**
+  * Registers a metric to be used to later update or call finish method
+  * @param {string} eventName name of the metric
+  * @param {string} id unique identifier
+  * @param {Object} props to populate the metric with
+  * @returns {bool} true if metric registered, false otherwise
```

Comment:
Is this bool expected to provide value outside of tests? If not, a spy may be more appropriate.

Reply:
The intention here is for the bool to potentially be used by whatever consumer is requesting a metric be registered. This could be used to determine if there is some problem with the metric registration/deletion process since this will be managed by way of register/update/send methods.

In the event a metric has already been registered with that ID, the consumer could read the FALSE return, and take some action (console log, send failure metric, secondary hashing, etc.).